src/persistence/settingsserializer.cpp

```diff
@@ -180,9 +188,12 @@ const SettingsSerializer::Value* SettingsSerializer::findValue(const QString& ke
         {
             if (a.group != group)
                 continue;
-            for (int vi : a.values)
-                if (values[vi].arrayIndex == arrayIndex && values[vi].key == key)
-                    return &values[vi];
+            for (int vi = 0; vi < a.values.size(); vi++)
+            {
+                const Value& v = values[vi];
+                if (v.arrayIndex == arrayIndex && v.key == key)
+                    return &v;
+            }
```

Comment:
>What was wrong with the simple range-for loop?

The `int` type is simply wrong. Would be a `Value` struct. Now how do you get the index, which is used to retrieve the value from the `values` array for comparison. Note: I'm not fixing/changing code architecture, just the type cast warnings!

And, uhm… how does that change meaning? Maybe I overlooked something, but it's the same, just done right. If however there is a modern way to retreive the container index, I'll gladly adopt it.